### ISSUE TITLE
MultipleChooserPanel should not allow selecting duplicate objects

### ISSUE THREAD

**Issue by pySilver** [wagtail/wagtail#10496]
### Issue Summary

it looks like `MultipleChooserPanel` allows selecting the same objects when accessed multiple times. Yes, it shows an appropriate error message on save but such behavior should not be possible in the first place or at least configurable.

**Comment by pySilver** [comment]
Good point. State issues :) What about passing the selection state back to the modal every time it's requested and handle unique optional requirement there.

**Comment by TonisPiip** [comment]
I am also facing this issue, and ask if there is some known way to pass in a QS of options into the MultipleChooserPanel so that the choices in the modal are limited with some logic, either by unique to what's already saved in the model, or some other logic.

**Comment by TonisPiip** [comment]
It seems that currently the most sane solution would be to get the reffer header from the request, and use that to reverse the url, get the object which is being edited, and then filter the QS based on that. IT doesn't include the current form state, but works well when adding a extra item to a unique list, as it'll only show valid non-duplciate rows.

**Comment by gasman** [comment]
The referer header is not reliable - browsers often block it for privacy reasons. However, passing a list of currently-selected object IDs to the chooser is not really the hard part - the hard part is keeping track of the currently selected items while navigating through the chooser.

**Comment by TonisPiip** [comment]
> currently selected items 

Meaning: the current state of the model Form? 

or

The currently edited object ID, is hard to track when navigating? 


Even having the currently edited object in the request, would be a massive improvement.
